### ISSUE TITLE
Firmware has errors

### ISSUE THREAD

**Issue by jcst40** [ANYCUBIC-3D/Kobra2#1]
This firmware has the following issues:
- Auto bed leveling skipping the initial steps, completely ignoring the button and silicone pad at the back.
- The settings button during prints only controls the fan, with no other functionality.
- Stop or pause print button not working.
- The Cooling fan can't be regulated. Works either 0% or 100%
- Can't resume prints when using M600

**Comment by yellobello** [comment]
Yes, I´ve looked at the source, and it seems like there is no code for the z-offset compensation probe anywhere to be found. Furthermore, it seems like it is the Vyper firmware, or at least the Vyper firmware was used as a base, but as I said I could not find any code for the z-offset probe.
Haven´t tried it on my printer yet, but I think I won´t even bother to upload it ... hopefully they will correct it. 
I´d love to use linear advance and input shaping, but I guess we will have to wait for the real firmware. This is the Vyper FW.

**Comment by marcokitt2000** [comment]
[Kobra 2_firmware_V2.8.7_EN.zip](https://github.com/ANYCUBIC-3D/Kobra2/files/11886331/Kobra.2_firmware_V2.8.7_EN.zip)
Hello Anycubic,

Where is the full open source firmware this is when compiled 167kb bin file. But on Facebook someone has the firmware 2.8.7. with the latest ui software (where a key option resume factory reset) its from 20230529T the binfile is now 478kb + hc32boot.bin 32kb mmmm that's 300kb less.
Please update this where auto-leveling option where the wipe is done en than module calibration is missing on this firmware.
And why M600 not working is maybe some fail in the marlin 2.0.x. bugfix old from last year.
Please give all info about it maybe users can fix for anycubic (that's is why open-source is).

But why you used old bugfix 2.0.x and why not used the 2.1.x or the latest 2.1.2.1.

Greetings Marco Kersten

**Comment by yellobello** [comment]
Hi Marco!
Does the Update that you posted include the z-offset calibration?

Dear Anycubic,

Please update the "sources" that you posted, since this is not the real source code. According to the GPL V3 you are required to do so, as you know and have been told countless times. You profit a lot from open source work and this is the least that you can do.

Best regards!

**Comment by marcokitt2000** [comment]
The zip file is orginal firmware with z offset calibration.

**Comment by marcokitt2000** [comment]
> Dear Anycubic,
> 
> Please update the "sources" that you posted, since this is not the real source code. According to the GPL V3 you are required to do so, as you know and have been told countless times. You profit a lot from open source work and this is the least that you can do.
> 
> Best regards!

This is on face book.
https://www.facebook.com/groups/anycubickobra2printer/posts/234861749399362/?comment_id=235429692675901&reply_comment_id=236765282542342&notif_id=1688056489799790&notif_t=group_comment

Someone has build it and updated and make a ticket to anycubic they sent it to him.

There are 2 version CH en EN both same but on ch is a other screen firmware 20230330T and on en is a updated 20230530t there is a key option resume factory setting mmm. but on here on de code there is  app map check that autoGetZoffseth and cpp and on inc marlinpreconfig is 

#if ENABLED(LEVEING_CALIBRATION_MODULE)
  //#include "autoGetZoffset.h" i have this enable itt
#endif

(LEVEING_CALIBRATION_MODULE where the hack this coming from i dont find it. and there is L is missing in Leveing mmmmm
when this is enable maybe it will work.
I think they have the wrong source online it was in build status before the printer was finish. that is what i mean the bin file is 170kb and ther others are 300kb more 470kb mmm there is much code missing.

I hope they will it do fast. or come with the latest marlin 2.1.2.1 build.
